src/Psy/Readline/HoaConsole.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2015 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy\Readline;
+
+use Hoa\Console\Console;
+use Hoa\Console\Readline\Readline as HoaReadline;
+use Psy\Exception\BreakException;
+
+/**
+ * Hoa\Console Readline implementation.
+ */
+class HoaConsole implements Readline
+{
+    /** @var HoaReadline */
+    private $hoa_readline;
+
+    /** @var \ReflectionProperty */
+    private $reflection;
+
+    /**
+     * @return bool
+     */
+    public static function isSupported()
+    {
+        return class_exists('\Hoa\Console\Console', true);
+    }
+
+    public function __construct()
+    {
+        $this->hoa_readline = new HoaReadline();
+        $this->reflection = new \ReflectionProperty($this->hoa_readline, '_history');
```

Comment:
Can we poke Hoa to get a proper API for history access? I'd rather not depend on reflection on a private member :(